### PR TITLE
Add an opt-in feature that saves a session per git branch (if you are in a git repo).

### PR DESCRIPTION
**Pull request by WhoIsSethDaniel**

### CODE REVIEW

**Comment by WhoIsSethDaniel** [comment]
bare-bones effort to get session per git branch.

File: lua/auto-session.lua
Comment (akinsho) [inline lua/auto-session.lua:96]: Just a drive by comment but this function `systemlist` is synchronous and will block nvim till it completes so if this is run at startup it will slow users startups down. An alternative is to use `jobstart` and pass a callback to receive the result, but then the whole outer function will have to adapt to use callbacks as well

Comment (WhoIsSethDaniel) [reply lua/auto-session.lua:96]: Yes, I'm aware. Redoing / adapting auto-session to use jobs isn't going to happen. At least not by me. Sorry.

Comment (akinsho) [reply lua/auto-session.lua:96]: @WhoIsSethDaniel no worries, fully a drive by comment, just pointing it out in case you weren't aware

Comment (WhoIsSethDaniel) [reply lua/auto-session.lua:96]: may be an idea for #116.  I wonder if the rewrite would benefit from using [plenary](https://github.com/nvim-lua/plenary.nvim). It has excellent job support.

Comment (rmagatti) [reply lua/auto-session.lua:96]: Good input, this is something I've thought about a bit, I'm just not sure if the core restoring and saving a session behaviour could actually be offloaded to a job. 🤔 As far as this git functionality, it's a good use-case for a job I'd say. 

I also agree @WhoIsSethDaniel that it's a feature better suited for the rewrite. I'll keep this in mind when working on it.

Thanks for chiming in @akinsho !

**Comment by rmagatti** [comment]
I'll try to test this out sometime this weekend, will merge if it looks good then!

**Review by rmagatti — APPROVED** [review]
Took me a while to get to this but this works well 👍

**Comment by rmagatti** [comment]
@WhoIsSethDaniel I'll try and add some documentation about this feature in the readme before merging, or you can do it if you happen to have a free minute. Either way, nice work, thanks for the contribution! 🎉

**Comment by WhoIsSethDaniel** [comment]
> @WhoIsSethDaniel I'll try and add some documentation about this feature in the readme before merging, or you can do it if you happen to have a free minute. Either way, nice work, thanks for the contribution! tada

I may have time tonight. Thanks for the review.

**Comment by WhoIsSethDaniel** [comment]
I added the new config element to README.md

**Pull request merged**
